core/baseline-core-v1.0-psd01.md

```diff
@@ -191,8 +253,286 @@ The ability of a Party operating Workflows on a baseline-compliant implementatio
 
 # 2 Design and Architecture
 
+This section provides definitions, key concepts, and overviews of the components of a Baseline Protocol Implementation compliant with the requirements of this document. 
+
+## 2.1 Agreement
+An agreement is a manifestation of mutual assent by two or more parties to one another see [Cornell Law School](https://www.law.cornell.edu/wex/agreement). An Agreement between two or more Counterparties allows for transactions between them dealing with, for example, commercial items such as products. An Agreement governs and defines all transactions between counterparties.
+
+**[R1]**	Transacting counterparties  MUST have a corresponding agreement.
+
+## 2.2	State Object
```

Comment:
This is a very useful object. Is it also reflected on the blockchain? (notarized) If yes, then is it a `must` or `may`?

Reply:
@mehranshakeri ...  yes state objects are reflected on the DLT through state proofs.